magmaR/R/retrieve.R

```diff
@@ -0,0 +1,168 @@
+
+retrieve <- function(
+    projectName,
+    modelName,
+    recordNames = "all",
+    attributeNames = "all",
+    filter = "",
+    page = NULL,
+    pageSize = 10,
+    connected.only = TRUE,
+    token = .get_TOKEN(),
+    ...
+) {
+    .retrieve(
+        projectName = projectName,
+        modelName = modelName,
+        recordNames = recordNames,
+        attributeNames = attributeNames,
+        filter = filter,
+        format = "tsv",
+        page = page,
+        pageSize = pageSize,
+        connected.only = TRUE,
+        token = token,
+        ...)
+}
+
+#' Download data from magma as a json, output as a json string (list)
+#' @inheritParams retrieve
+#' @param filter potential filter of the data
+#' @return a json string or list of such strings
+#' @export
+#' @examples
+#' 
+#' # Unless a working TOKEN is hard-coded, or it is in an interactive session,
+#' #   this code will not work.
+#' 
+#' if (interactive()) {
+#'     # Running like this will ask for input of your janus token.
+#'     retrieveJSON(
+#'         projectName = "ipi",
+#'         modelName = "patient",
+#'         recordNames = "all",
+#'         attributeNames = "all",
+#'         filter = "")
+#' }
+#' 
+retrieveJSON <- function(
+    projectName,
+    modelName,
+    recordNames = "all",
+    attributeNames = "all",
+    filter = "",
+    page = NULL,
+    pageSize = 10,
+    token = .get_TOKEN(),
+    ...
+) {
+    
+    .retrieve(
+        projectName = projectName,
+        modelName = modelName,
+        recordNames = recordNames,
+        attributeNames = attributeNames,
+        filter = filter,
+        format = "json",
+        page = page,
+        pageSize = pageSize,
+        token = token,
+        ...)
+}
+
+.retrieve <- function(
+    projectName,
+    modelName,
+    recordNames,
+    attributeNames,
+    format = c("tsv", "json"),
+    filter = "",
+    page = NULL,
+    pageSize = 10,
+    connected.only = TRUE,
+    names.only = FALSE,
+    request.only = FALSE,
+    json.params.only = FALSE,
+    token = .get_TOKEN(),
+    verbose = FALSE
+) {
+    
+    format <- match.arg(format)
+    
+    ### Put together the request 
+    
+    jsonParams <- list(
+        project_name = projectName,
+        model_name = modelName,
+        record_names = .as_array_unless_all_or_brackets(recordNames),
+        attribute_names = .as_array_unless_all_or_identifier(attributeNames),
+        filter = filter,
+        format = format)
+
+    if (!identical(page, NULL)) {
+        jsonParams$page <- page
+        jsonParams$page_size <- pageSize
+    }
+    
+    requestBody <- jsonlite::toJSON(jsonParams, auto_unbox = TRUE)
+    
+    ### Output here if requested.
+    if (request.only) {
+        return(requestBody)
+    }
+    if (json.params.only) {
+        return(jsonParams)
+    }
+    
+    ### Perform '\retrieve'-al
+    
+    h <- RCurl::basicTextGatherer()
+    
+    RCurl::curlPerform(
+        url = "https://magma.ucsf.edu/retrieve",
```

Comment:
It might be useful for testing if this value could be configurable in a non-committed / `.gitignore` config file.